components/eventing-controller/reconciler/subscription/reconciler_test.go

```diff
@@ -762,6 +762,98 @@ var _ = Describe("Subscription Reconciliation Tests", func() {
 		})
 	})
 
+	When("Deleting BEB Subscription manually", func() {
+		It("Should recreate BEB Subscription again", func() {
+			kymaSubscriptionName := "test-subscription"
+			kymaSubscription := reconcilertesting.NewSubscription(kymaSubscriptionName, namespaceName,
+				reconcilertesting.WithWebhookAuthForBEB, reconcilertesting.WithEventTypeFilter)
+
+			Context("Setup Kyma Subscription required resources", func() {
+				var svc *v1.Service
+				By("Creating Subscriber service", func() {
+					svc = reconcilertesting.NewSubscriberSvc("test-service", namespaceName)
+					ensureSubscriberSvcCreated(svc, ctx)
+				})
+
+				By("Creating Kyma Subscription", func() {
+					reconcilertesting.WithValidSink(svc.Namespace, svc.Name, kymaSubscription)
+					ensureSubscriptionCreated(kymaSubscription, ctx)
+				})
+
+				By("Creating APIRule", func() {
+					getAPIRuleForASvc(svc, ctx).Should(reconcilertesting.HaveNotEmptyAPIRule())
+				})
+
+				By("Updating APIRule status to be Ready", func() {
+					apiRule := filterAPIRulesForASvc(getAPIRules(ctx, svc), svc)
+					ensureAPIRuleStatusUpdatedWithStatusReady(&apiRule, ctx).Should(BeNil())
+				})
+			})
+
+			Context("Check Kyma Subscription ready", func() {
+				By("Checking BEB mock server creation requests to contain Subscription creation request", func() {
+					Eventually(func() bool {
+						for r, payload := range beb.Requests {
+							if reconcilertesting.IsBebSubscriptionCreate(r, *beb.BebConfig) {
+								if bebSubscription, ok := payload.(bebtypes.Subscription); !ok {
+									return false
+								} else {
+									return kymaSubscriptionName == bebSubscription.Name
+								}
+							}
+						}
+						return false
+					}).Should(BeTrue())
+				})
+
+				By("Checking Kyma Subscription ready condition to be true", func() {
+					getSubscription(kymaSubscription, ctx).Should(And(
+						reconcilertesting.HaveSubscriptionName(kymaSubscriptionName),
+						reconcilertesting.HaveSubscriptionReady(),
+					))
+				})
+			})
+
+			notfoundHandlerCalled := make(chan bool, 1)
+			Context("Simulate manual deletion of BEB Subscription", func() {
```

Comment:
I think we should not do this (playing with the Mock behaviour). The test should only rely on asserting simple REST calls that are made to the mock (ideally we have some stubs auto-generated from the OpenAPI spec of BEB). If the test needs more behaviour than simple call expectations, then we shouldn't use a mock, since it will be just relyinhg on some logic that we wrote and might not represent the actual system being tested.

Reply:
Since this is not the main focus of this ticket, could you please create a ticket for this?

Reply:
Since this is not the main focus of this ticket, could you please create a ticket for it?